cachi2/core/package_managers/npm.py

```diff
@@ -217,22 +244,35 @@ def get_package_name_from_path(package_path: str) -> str:
 
         return main_package, packages
 
-    def get_main_package(self) -> dict[str, str]:
+    def get_main_package(self) -> NpmComponentInfo:
         """Return a dict with sbom component data for the main package."""
         name = self._lockfile_data["name"]
         version = self._lockfile_data["version"]
         purl = self._purlifier.get_purl(name, version, "file:.", integrity=None)
         return {"name": name, "version": version, "purl": purl.to_string()}
 
-    def get_sbom_components(self) -> list[dict[str, Optional[str]]]:
+    def get_sbom_components(self) -> list[NpmComponentInfo]:
         """Return a list of dicts with sbom component data."""
         packages = self._packages
 
-        def to_component(package: Package) -> dict[str, Optional[str]]:
-            name = package.name
-            version = package.version
-            purl = self._purlifier.get_purl(name, version, package.resolved_url, package.integrity)
-            return {"name": name, "version": version, "purl": purl.to_string()}
+        def to_component(package: Package) -> NpmComponentInfo:
+            purl = self._purlifier.get_purl(
+                package.name, package.version, package.resolved_url, package.integrity
+            ).to_string()
+            component: NpmComponentInfo = {
+                "name": package.name,
+                "version": package.version,
+                "purl": purl,
+            }
+
+            for key, value in [
+                ("dev", package.dev),
+                ("bundled", package.bundled),
+            ]:
+                if value:
+                    component[key] = value  # type: ignore[literal-required]
```

Comment:
while `NotRequired` is quite fancy, maybe making them required and passing the explicit `False` would result in fewer type-ignores?